src/toc.ts

```diff
@@ -470,25 +484,83 @@ export function getAllRootHeading(doc: TextDocument): IHeading[] {
             // The magic comment is at the end of the heading.
             || crtLineText.endsWith('<!-- omit in toc -->')
             || crtLineText.endsWith('<!-- omit in TOC -->')
-        ) {
-            isInToc = false;
+        )) {
+            entry.isInToc = false;
         }
 
-        // Extract heading info.
-        const matches = /^ {0,3}(#{1,6})(.*)$/.exec(crtLineText)!;
-        const entry = {
-            level: matches[1].length as MarkdownHeadingLevel,
-            rawContent: matches[2].replace(/^[ \t]+/, '').replace(/[ \t]+#+[ \t]*$/, ''),
-            lineIndex: i,
-            isInToc,
-        };
+        if (respectProjectLevelOmit
+            && entry.isInToc
+            && projectLevelOmittedHeadings.some(({ level, text }) => level === entry.level && text === entry.rawContent)
+        ) {
+            entry.isInToc = false;
+        }
 
         toc.push(entry);
     }
 
     return toc;
 }
 
+/**
+ * Gets all headings in the root of the text document, with additional TOC specific properties.
+ * @returns In ascending order of `lineIndex`.
+ */
+export function getAllTocEntry({
+    doc,
+    respectMagicCommentOmit = false,
+    respectProjectLevelOmit = false,
+    slugifyMode = workspace.getConfiguration('markdown.extension.toc').get<SlugifyMode>('slugifyMode')!,
+}: {
+    doc: TextDocument;
+    respectMagicCommentOmit?: boolean;
+    respectProjectLevelOmit?: boolean;
+    slugifyMode?: SlugifyMode;
+}): ITocItem[] {
```

Comment:
What is the purpose of passing a param object? type check?

Reply:
`getAllTocEntry()` accepts **a good number of optional parameters**. Then, I wanna make their order flexible, so that don't have to change callers every time making tiny changes to it.

I do expect **frequent** changes to the signature of `getAllTocEntry()`.

Reply:
Fair enough. Or we can at least make `doc` positional and the rest as a "config" object, just like many VS Code APIs do.